### PR TITLE
Add resources for groups, service accounts, and project memberships

### PR DESCRIPTION
**Pull request by nmanoogian**
This PR adds resources for groups, service accounts, and project memberships.

Project memberships associate groups and service accounts with Doppler projects. Memberships have an associated project role and a list of environments.

Closes ENG-4997

### CODE REVIEW

**Comment by nmanoogian** [comment]
Force pushed to remove user project member resource

File: examples/resources/project_member_group.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+resource "doppler_project_member_group" "backend_engineering" {
+  project      = "backend"
+  group_slug   = doppler_group.engineering.slug
+  role         = "collaborator"
+  environments = ["dev", "stg"]
```
Comment (Piccirello) [inline examples/resources/project_member_group.tf:5]: nit: in this example, should the `project` and `environments` also reference a terraform resource? (like the `group_slug`)

File: doppler/models.go [hunk 2]
```diff
@@ -207,3 +207,23 @@ func parseServiceTokenResourceId(id string) (project string, config string, slug
...
+	IsInlineRole bool     `json:"is_inline_role"`
+	CreatedAt    string   `json:"created_at"`
+}
+
+type ServiceAccount struct {
+	Slug          string                      `json:"slug"`
+	Name          string                      `json:"name"`
+	CreatedAt     string                      `json:"created_at"`
+	WorkplaceRole ServiceAccountWorkplaceRole `json:"workplace_role"`
+}
```
Comment (Piccirello) [inline doppler/models.go:225]: On the backend we have a single WorkplaceRoleDTO for service accounts, users, invites, etc. It's the same DTO we use for our actual workplace role endpoint (which we'll add a terraform resource for in the future). You can thus have a single `WorkplaceRole` type if you'd prefer, but totally up to you.

**Review by Piccirello — APPROVED** [review]

File: doppler/models.go (hunk 2 above)
Comment (Piccirello) [reply doppler/models.go:225]: Precisely - just that we'll be able to use this more broadly when we inevitable add additional resources that contain a workplace role.

File: examples/resources/project_member_group.tf (hunk 1 above)
Comment (Piccirello) [reply examples/resources/project_member_group.tf:5]: That SGTM!

File: doppler/models.go (hunk 2 above)
Comment (nmanoogian) [reply doppler/models.go:225]: Sounds good! I'll make the change

**Comment by nmanoogian** [comment]
[Force push](https://github.com/DopplerHQ/terraform-provider-doppler/compare/bc4093d6d166554a8ec3b251c22f413ffc989bae..4a9c048b7255a0806ecee323217b626ddc569c38) renames `ServiceAccountWorkplaceRole` to `WorkplaceRole`

**Review by seslattery — APPROVED** [review]
I think this looks good overall. I don't love the `payload := map[string]interface{}{}` pattern we're using, but it's consistent with the rest of the project.  We lose some type safety and duplicate effort compared to using something like an input struct, but this is fine and working.

**Comment by nmanoogian** [comment]
> I think this looks good overall. I don't love the `payload := map[string]interface{}{}` pattern we're using, but it's consistent with the rest of the project. We lose some type safety and duplicate effort compared to using something like an input struct, but this is fine and working.

Awesome, thanks! I agree completely, we definitely want to be moving in the direction of input structs

**Pull request merged**